extensions/amp-bind/0.1/bind-impl.js

```diff
@@ -170,12 +183,74 @@ export class Bind {
       dev().fine(TAG, `Initialized ${bindings.length} bindings from ` +
           `${boundElements.length} elements.`);
     });
+    return this.scanPromise_;
+  }
+
+  /**
+   * Removes all bindings nodes with `node` as their parent.
+   *
+   * Returns a promise that resolves after bindings have been removed.
+   *
+   * @param {!Element} node
+   * @return {Promise}
+   *
+   * @private
+   */
+  removeBindingsForNode_(node) {
+    return new Promise(resolve => {
+      // Eliminate bound elements that have node as an ancestor
+      for (let i = this.boundElements_.length - 1; i >= 0; i--) {
```

Comment:
We have a [`filterSplice`](https://github.com/ampproject/amphtml/blob/master/src/utils/array.js#L26) helper for this.